scrapyd/environ.py

```diff
@@ -30,11 +32,39 @@ def get_environment(self, message, slot):
         if project in self.settings:
             env['SCRAPY_SETTINGS_MODULE'] = self.settings[project]
         if self.logs_dir:
-            env['SCRAPY_LOG_FILE'] = self._get_file(message, self.logs_dir, 'log')
+            env['SCRAPY_LOG_FILE'] = self._get_log_file(message)
         if self.items_dir:
             env['SCRAPY_FEED_URI'] = self._get_feed_uri(message, 'jl')
         return env
 
+    def _get_log_file(self, message):
+        """Combine the logs_dir and logs_filename config items, and substitute
+        any variables in logs_filename to get the full filename of the log file
+        """
+        if not self.logs_filename:
+            # if no filename specified, fall back on the default.
+            return self._get_file(message, self.logs_dir, 'log')
+
+        now = datetime.datetime.now()
+        format_args = {}
+        format_args['project'] = message['_project']
+        format_args['spider'] = message['_spider']
+        format_args['Y'] = now.year
+        format_args['m'] = now.month
+        format_args['d'] = now.day
+        format_args['H'] = now.hour
+        format_args['M'] = now.minute
+        format_args['S'] = now.second
+        filename = self.logs_filename.format(**format_args)
+
+        full_filename = os.path.join(self.logs_dir, filename)
+
+        log_dir = os.path.dirname(full_filename)
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
```

Comment:
* Changing the permission is a good idea. I think that `744` is a good default setting.
* Again, we can catch the exception, but what should we do with it?
* There is a note about handling the log_dir when log_filename is set in the documentation
Is there much else we need? https://github.com/sujaymansingh/scrapyd/commit/5cdbb752199c0557e7530e0e3cdb673f3b5efe9b#diff-fd40cf2be7711772de9d8316da038cceR71